src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1216,3 +1238,23 @@ impl<T: 'static> EventProcessor<T> {
         }
     }
 }
+
+fn is_first_touch(first: &mut Option<u64>, num: &mut u32, id: u64, phase: TouchPhase) -> bool {
+    match phase {
+        TouchPhase::Started => {
+            if *num == 0 {
+                *first = Some(id);
+            }
+            *num += 1;
+        }
+        TouchPhase::Cancelled | TouchPhase::Ended => {
+            if *first == Some(id) {
+                *first = None;
```

Comment:
My understanding is that this is correct, but unnecessary: if `num == 0` then `first == Some(id)` is never evaluated (except on `TouchPhase::Ended` for the last event, when the above ids are still valid). As soon as the next event starts, `first` is re-assigned. Thus in my version I didn't bother using an `Option` here.